[TIMOB-26885] fix(ios): prevent VC from being pushed twice

hansemannn opened the pull request:
**JIRA:** https://jira.appcelerator.org/browse/TIMOB-26885

See JIRA for details. Not unit-testable since only happening on very old devices and only very rarely.

hansemannn committed:
[TIMOB-26885] fix(ios): prevent VC from being pushed twice

build commented:
<!--
  2 failure:  :microscope: Ther..., :disappointed_rel...
  1 warning:  :mag: Can't find ...
  2 messages
  
  DangerID: danger-id-default;
-->

<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  :microscope: There are library changes, but no changes to the unit tests. That's OK as long as you're refactoring existing code, but will require an admin to merge this PR. Please see <a href="https://github.com/hansemannn/titanium_mobile/blob/TIMOB-26885/README.md#unit-tests">README.md#unit-tests</a> for docs on unit testing.
  </td>
    </tr>
  
<tr>
      <td>:no_entry_sign:</td>
      <td>

  :disappointed_relieved: `npm test` failed. See below for details.
  </td>
    </tr>
  </tbody>
</table>


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  :mag: Can't find junit reports at `./junit.*.xml`, skipping generating JUnit Report.
  </td>
    </tr>
  </tbody>
</table>


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Messages</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:book:</td>
      <td>:tada: Another contribution from our awesome community member, hansemannn! Thanks again for helping us make Titanium SDK better. :thumbsup:</td>
    </tr>
  
<tr>
      <td>:book:</td>
      <td>

  ```
> titanium-mobile@8.1.0 test /Users/build/jenkins/workspace/ium-sdk_titanium_mobile_PR-10760
> npm run ios-sanity-check && grunt


> titanium-mobile@8.1.0 ios-sanity-check /Users/build/jenkins/workspace/ium-sdk_titanium_mobile_PR-10760
> ./build/scons check-ios-toplevel

Running "appcJs:src:lintOnly" (appcJs) task

Running "eslint:src" (eslint) task

/Users/build/jenkins/workspace/ium-sdk_titanium_mobile_PR-10760/build/scons-xcode-test.js
  27:3  warning  Each then() should return a value or throw  promise/always-return

/Users/build/jenkins/workspace/ium-sdk_titanium_mobile_PR-10760/build/utils.js
  154:3   warning  Avoid using promises inside of callbacks    promise/no-promise-in-callback
  154:3   warning  Avoid using promises inside of callbacks    promise/no-promise-in-callback
  154:63  warning  Each then() should return a value or throw  promise/always-return
  155:4   warning  Avoid calling back inside of a promise      promise/no-callback-in-promise
  157:4   warning  Avoid calling back inside of a promise      promise/no-callback-in-promise
  175:59  warning  Each then() should return a value or throw  promise/always-return
  176:4   warning  Avoid calling back inside of a promise      promise/no-callback-in-promise
  178:4   warning  Avoid calling back inside of a promise      promise/no-callback-in-promise
  189:3   warning  Avoid using promises inside of callbacks    promise/no-promise-in-callback
  189:3   warning  Avoid using promises inside of callbacks    promise/no-promise-in-callback
  189:51  warning  Each then() should return a value or throw  promise/always-return
  190:4   warning  Avoid calling back inside of a promise      promise/no-callback-in-promise
  192:4   warning  Avoid calling back inside of a promise      promise/no-callback-in-promise
  203:71  warning  Each then() should return a value or throw  promise/always-return
  205:4   warning  Avoid calling back inside of a promise      promise/no-callback-in-promise
  206:17  warning  Avoid calling back inside of a promise      promise/no-callback-in-promise
  218:71  warning  Each then() should return a value or throw  promise/always-return
  220:4   warning  Avoid calling back inside of a promise      promise/no-callback-in-promise

/Users/build/jenkins/workspace/ium-sdk_titanium_mobile_PR-10760/common/Resources/ti.internal/bootstrap.loader.js
  122:17  warning  Found non-literal argument in require  security/detect-non-literal-require

/Users/build/jenkins/workspace/ium-sdk_titanium_mobile_PR-10760/iphone/cli/commands/_build.js
  6410:79  warning  'out' is defined but never used  no-unused-vars

/Users/build/jenkins/workspace/ium-sdk_titanium_mobile_PR-10760/iphone/cli/hooks/frameworks.js
   61:5   warning  Expected catch() or return                  promise/catch-or-return
   61:34  warning  Avoid calling back inside of a promise      promise/no-callback-in-promise
  428:5   warning  Each then() should return a value or throw  promise/always-return

✖ 24 problems (0 errors, 24 warnings)


Running "checkFormat:ios" (checkFormat) task
Fatal error: Error: Formatting incorrect on "iphone/Classes/TiUITabProxy.m", proposed changes: <?xml version='1.0'?>
<replacements xml:space='preserve' incomplete_format='false'>
<replacement offset='3803' length='2'> </replacement>
</replacements>

npm ERR! Test failed.  See above for more details.

```
  </td>
    </tr>
  </tbody>
</table>


<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 23a7991313d6813f946815d5d5bd3d195cfbd9c7
</p>

sgtcoolguy committed:
Merge branch 'master' into TIMOB-26885

hansemannn committed:
fix(lambus): properly guard window pushes

# Conflicts:
#	iphone/Classes/TiUINavigationWindowProxy.m
#	iphone/Classes/TiUITabProxy.m

vijaysingh-axway committed:
Merge branch 'master' into TIMOB-26885

vijaysingh-axway reviewed (approved):
CR passed.

vijaysingh-axway committed:
Merge branch 'master' into TIMOB-26885

build commented:
<!--
  0 failure: 
  0 warning: 
  2 messages
  
  DangerID: danger-id-Danger;
-->



<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Messages</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:book:</td>
      <td>:tada: Another contribution from our awesome community member, hansemannn! Thanks again for helping us make Titanium SDK better. :thumbsup:</td>
    </tr>
  
<tr>
      <td>:book:</td>
      <td>

  :white_check_mark: All tests are passing
Nice one! All 1970 tests are passing.
  </td>
    </tr>
  </tbody>
</table>


<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 4edddaee1359ed3d84746889feb25da112556b5f
</p>

keerthi1032 commented:
FR Passed. Tested using Kitchen-sink app on older simulator and device.No crash occurs.
Operating System
  Name                        = Mac OS X
  Version                     = 10.13.6
  Architecture                = 64bit
Node.js
  Node.js Version             = 8.9.1
  npm Version                 = 5.5.1
Titanium CLI
  CLI Version                 = 5.1.1
Titanium SDK
  SDK Version                 = local sdk 8.1.0.v20190418111302
  Device  =iPhone X iOS 11,iphone XR iOS 12
Simulator = iPhone 6 iOS 11. iPhone 5s iOS 10

keerthi1032 committed:
Merge branch 'master' into TIMOB-26885

keerthi1032 committed:
Merge branch 'master' into TIMOB-26885
